core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -115,20 +115,32 @@ public void run() {
   }
 
   @Override
-  public synchronized ClientStream newStream(MethodDescriptor<?, ?> method,
-      Metadata headers, ClientStreamListener clientStreamListener) {
+  public synchronized ClientStream newStream(
+      final MethodDescriptor<?, ?> method, final Metadata headers) {
     if (shutdownStatus != null) {
-      clientStreamListener.closed(shutdownStatus, new Metadata());
-      return new NoopClientStream();
+      return new NoopClientStream() {
+        @Override
+        public void start(ClientStreamListener listener) {
+          listener.closed(shutdownStatus, new Metadata());
```

Comment:
Updated docs.  I don't think it is feasibly to not call the listener from start.